tests/dags/providers/provider_api_scripts/test_smithsonian.py

```diff
@@ -59,6 +59,17 @@ def test_alert_new_unit_codes():
     )
 
 
+def test_get_unit_codes_from_api_retries():
+    mock_response = {"terms": []}
+    # Patch the sleep function so it doesn't take long
+    with patch.object(
+        si.delayed_requester, "get_response_json", return_value=mock_response
+    ) as request_patch, patch("time.sleep"):
+        with pytest.raises(ValueError, match="No unit codes received."):
+            si.get_unit_codes_from_api()
+            assert request_patch.call_count == 3
```

Comment:
I think if `si.get_unit_codes_from_api()` raises a `ValueError`, then the assert statement won't be reached right? Maybe it should be outside the `pytest.raises` context so it occurs after the `ValueError` is caught?

Reply:
Yes, Madison is correct:

https://github.com/pytest-dev/pytest/blob/main/src/_pytest/python_api.py#L838-L848

Reply:
Oops, that's right of course -- oversight, fixed :)